tests/scalers/azure-data-explorer.test.ts

```diff
@@ -5,6 +5,7 @@ import test from 'ava'
 
 const dataExplorerDb = process.env['AZURE_DATA_EXPLORER_DB']
 const dataExplorerEndpoint = process.env['AZURE_DATA_EXPLORER_ENDPOINT']
+const msiClientId = process.env['AZURE_MSI_CLIENT_ID']
```

Comment:
We should add this env variable to worklflows under `.github/workflows`, eg. https://github.com/kedacore/keda/blob/13313241f012c1b8c4ad8fbdfc55e5cd617f005f/.github/workflows/pr-e2e.yml#L69

Reply:
@JorTurFer @tomkerkhove we need to add this variable to GH secrets

Reply:
As far as I can see, we don't have AAD-Pod-Identity deployed in the clusters (nor in nightly nor pr) and I cannot do it. 
If we need it, @tomkerkhove or @ahmelsayed have to do it and create the proper identities

Reply:
> As far as I can see, we don't have AAD-Pod-Identity deployed in the clusters (nor in nightly nor pr) and I cannot do it. If we need it, @tomkerkhove or @ahmelsayed have to do it and create the proper identities

@JorTurFer - Are You sure you don't have aad-pod-identity in your e2e cluster? because one of the e2e scenarios of azure-data-explorer scaler is relying on that and it completed successfully until the [bug](https://github.com/kedacore/keda/issues/2770) this pr is going to fix